cmd/explored/main.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/types"
+	"go.sia.tech/siad/v2/api/explored"
+	"go.sia.tech/siad/v2/p2p"
+)
+
+var (
+	// to be supplied at build time
+	githash   = "?"
+	builddate = "?"
+)
+
+var (
+	genesisTxns  = []types.Transaction{}
+	genesisBlock = types.Block{
+		Header: types.BlockHeader{
+			Timestamp: time.Unix(734600000, 0),
+		},
+		Transactions: genesisTxns,
+	}
+	genesisUpdate = consensus.GenesisUpdate(genesisBlock, types.Work{NumHashes: [32]byte{29: 1 << 4}})
+	genesis       = consensus.Checkpoint{Block: genesisBlock, Context: genesisUpdate.Context}
+)
+
+func die(context string, err error) {
+	if err != nil {
+		log.Fatalf("%v: %v", context, err)
+	}
+}
+
+func main() {
+	log.SetFlags(0)
+	gatewayAddr := flag.String("addr", ":0", "address to listen on")
+	apiAddr := flag.String("http", "localhost:9980", "address to serve API on")
+	dir := flag.String("dir", ".", "directory to store node state in")
+	checkpoint := flag.String("checkpoint", "", "checkpoint to bootstrap from")
+	bootstrap := flag.String("bootstrap", "", "peer address or explorer URL to bootstrap from")
+	flag.Parse()
+
+	log.Println("explored v0.1.0")
+	if flag.Arg(0) == "version" {
+		return
+	}
+
+	initCheckpoint := genesis
+	if *checkpoint != "" {
```

Comment:
we can probably remove this functionality from the explorer; explorers should track the history of the entire chain, so it doesn't make much sense for them to bootstrap from a recent checkpoint